packages/graphql/server/resolvers/mutators.ts

```diff
@@ -149,11 +210,20 @@ export const createMutator = async <TModel extends VulcanDocument>({
     });
   }
 
+  /* Autorization */
+  if (!asAdmin) {
```

Comment:
I'd be tempted to move this check into the performMutationCheck.
So, we always run the check, but inside the check, we skip relevant steps if the user is an admin or we are using the "asAdmin" option.

Because performMutationCheck also do stuff like checking if the document actually exist. It checks permissions but also "validity" of the update operation. So you want to run some parts of it even when you are an admin.